pyprof/prof/reduction.py

```diff
@@ -44,44 +42,33 @@ def __init__(self, d):
 
         # The input can be a scalar or a tensor
         if 'shape' in i:  # tensor
-            self.shape = i['shape']
-            self.type = i['dtype']
+            self.inp = Tensor(i['shape'], i['dtype'])
         else:  # scalar
             assert ('value' in i)
-            self.shape = (1, )
-            self.type = i['type']
+            self.inp = Tensor([], i['type'])
 
         self.dir = d.dir
         self.sub = d.sub
 
     def params(self):
-        p = OrderedDict([('T', self.shape), ('type', self.type)])
-        return p
+        return str(self.inp)
 
     def tc(self):
         return "-"
 
     def op(self):
-        return self.op_
+        return self._op
```

Comment:
Our coding style is to typically have internal members end with `_`, not start. The former scheme was correct.